src/internal/CircleRipple.js

```diff
@@ -4,10 +4,26 @@ import shallowEqual from 'recompose/shallowEqual';
 import autoPrefix from '../utils/autoPrefix';
 import transitions from '../styles/transitions';
 
+// A helper function that calls back when any pending animations have started
+const requestAnimationStart = (cb) => {
```

Comment:
I think that we miss a story to handle that source of asynchronicity. My main concern is with component unmounted while having a pending `requestAnimationStart`.